rollup.config.js

```diff
@@ -0,0 +1,113 @@
+import commonjs from '@rollup/plugin-commonjs';
+import resolve from '@rollup/plugin-node-resolve';
+import replace from '@rollup/plugin-replace';
+import typescript from '@rollup/plugin-typescript';
+import babel from 'rollup-plugin-babel';
+import { terser } from 'rollup-plugin-terser';
+
+import pkg from './package.json';
+
+const extensions = ['.js', '.jsx', '.ts', '.tsx'];
+
+const external = [];
+
+if (pkg.peerDependencies) {
+  external.push(...Object.keys(pkg.peerDependencies));
+}
+
+if (pkg.dependencies) {
+  external.push(...Object.keys(pkg.dependencies));
+}
+
+const terserPretty = terser({
+  warnings: true,
+  ecma: 5,
+  keep_fnames: true,
+  compress: {
+    conditionals: false,
+    if_return: false,
+    join_vars: false,
+    keep_fnames: true,
+    loops: false,
+    pure_getters: true,
+    toplevel: true,
+    sequences: false,
+  },
+  mangle: false,
+  output: {
+    braces: true,
+    indent_level: 2,
+  },
+});
+
+const terserMinified = terser({
+  warnings: true,
+  ecma: 5,
+  toplevel: true,
+  compress: {
+    keep_infinity: true,
+    passes: 10,
+    pure_getters: true,
+  },
+});
+
+const makePlugins = isProduction =>
+  [
+    resolve({
+      mainFields: ['module', 'jsnext', 'main'],
+      browser: true,
+    }),
+    commonjs({
+      ignoreGlobal: true,
+      include: /\/node_modules\//,
+      namedExports: {
+        react: Object.keys(require('react')),
+      },
+    }),
+    typescript({
+      typescript: require('typescript'),
+    }),
+    babel({ extensions, include: ['src/**/*'], exclude: 'node_modules/**' }),
```

Comment:
The output that will come out is ES6 and won't be compatible some browsers (IE11 edge,...)  this because tsconfig has ES2015 target and babel config has current node

Reply:
Good catch! I didn't even notice that, but in looking at the non-minified build output now there definitely is some ES6 in there 😅 🆒 I'll give `microbundle` a shot!

Reply:
I think it’s also worth just copying what we have for urql and combining it with what we have in Wonka (it already looks pretty similar)

Happy to send some commits over for that!

Basically, we use TypeScript to compile to certain helpers but not much, we then use Buble because it handles classes and spreading/rest operators the best, lastly we use Babel only for the “detail” work, like making sure that we have a unified helper for `Object.assign`

We also recently added `closure-compiler` but that can mangle code which may not be a good choice for renature

Reply:
I'd love to have you send over some commits if you have the time! I did manage to get `microbundle` working with pretty minimal effort, but I'd like to see what you have in mine. Agreed that I don't think it's quite worth getting full closure compiler stuff in just yet, but everything else sounds good 😄